src/api/services/productServices.ts

```diff
@@ -17,14 +20,38 @@ export const createProductService = async (product: ProductDataInterface) => {
 };
 
 // Fetch All Products Services
-export const getAllProductsService = async () => {
-  const allProducts = await productModel.find({});
+export const getAllProductsService = async (
+  productModel: Model<ProductDataInterface>,
+  options: GetAllProductsOptions
+): Promise<ProductDataInterface[]> => {
+  // Sorting, limiting and pagination of the Products
```

Comment:
trying to implement sorting, pagination and limiting of the fields on this endpoint but cannot get it to work. Would need help on this!